### PR TITLE
add kuttl logs into scorecard output when running kuttl tests

### PR DESCRIPTION
**Pull request by jmccormick2001**
**Description of the change:**
this PR adds kuttl stderr/stdout logs output into the scorecard test result output when the scorecard-test-kuttl 
image is running tests.

**Motivation for the change:**
the kuttl logs are useful to debug kuttl test output, for example, kuttl tests require you to run scorecard with --service-account=somekuttlsa specifying a service account with RBAC that allows 'list' on namespaces, kuttl
will fail to run tests and its output logs are useful in debugging this scenario.

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

File: images/scorecard-test-kuttl/cmd/test-kuttl/main.go [hunk 1]
```diff
@@ -175,3 +188,33 @@ type Testsuites struct {
...
+		return err.Error()
+	}
+	defer stderrFile.Close()
+
+	var byteValue []byte
+	byteValue, err = ioutil.ReadAll(stderrFile)
+	if err != nil {
+		printErrorStatus(fmt.Errorf("could not read kuttl stderr file %v", err))
+		return err.Error()
+	}
```
Comment (joelanford) [inline images/scorecard-test-kuttl/cmd/test-kuttl/main.go:205]: These two calls could be simplified to one with [`ioutil.ReadFile`](https://golang.org/pkg/io/ioutil/#ReadFile).

Ditto for `kuttl.stdout`

File: images/scorecard-test-kuttl/cmd/test-kuttl/main.go [hunk 2]
```diff
@@ -74,9 +74,22 @@ func main() {
...
+		r := v1alpha3.TestResult{}
+		r.Log = logMessage
+		s.Results = append(s.Results, r)
+		return s
+	}
+
 	for i := 0; i < len(tc); i++ {
 		r := v1alpha3.TestResult{}
 		r.Name = tc[i].Name
+		r.Log = logMessage
```
Comment (joelanford) [inline images/scorecard-test-kuttl/cmd/test-kuttl/main.go:92]: This is putting the same log message in every single test result right? I'm not familiar with kuttl's verbosity when things run normally, but I'm wondering if we should only show the logs when `len(tc) == 0`?

Comment (jmccormick2001) [reply images/scorecard-test-kuttl/cmd/test-kuttl/main.go:92]: correct, it goes in every test result....kuttl has a single stderr and stdout for all its tests, we could only add that log output when len(tc) == 0 but then you lose the ability to view the logs altogether which can be useful.  I was wondering about adding a Log field to the scorecard API at the TestResult level to handle this sort of case but that requires a scorecard API change.

Comment (joelanford) [reply images/scorecard-test-kuttl/cmd/test-kuttl/main.go:92]: > you lose the ability to view the logs altogether which can be useful

If we did it this way, wouldn't it mean seeing the logs when the `kuttl` process fails and not seeing the logs when kuttl returns test case results? If so, that seems pretty reasonable at least as a first step.

Is the stdout or stderr useful to see when tests run normally, or only when kuttl bombs out?
Also, is there any possibility of parsing the kuttl's logs and splitting for each test case?

Comment (estroz) [reply images/scorecard-test-kuttl/cmd/test-kuttl/main.go:92]: Alternatively, will logs be different per test? Can these logs be collected by `tc[i].Name` and added to the appropriate result?

Comment (jmccormick2001) [reply images/scorecard-test-kuttl/cmd/test-kuttl/main.go:92]: ``
	Log:
		=== RUN   kuttl
		    harness.go:435: starting setup
		    harness.go:247: running tests using configured kubeconfig.
		    harness.go:335: running tests
		    harness.go:73: going to run test suite with timeout of 120 seconds for each step
		=== RUN   kuttl/harness
		=== RUN   kuttl/harness/list-pods
		=== PAUSE kuttl/harness/list-pods
		=== CONT  kuttl/harness/list-pods
		    logger.go:42: 20:37:15 | list-pods | Skipping creation of user-supplied namespace: default
		    logger.go:42: 20:37:15 | list-pods/0-pod | starting test step 0-pod
		    logger.go:42: 20:37:15 | list-pods/0-pod | Pod:default/pod-1 created
		    logger.go:42: 20:37:15 | list-pods/0-pod | test step completed 0-pod
		    logger.go:42: 20:37:15 | list-pods | skipping kubernetes event logging
		    logger.go:42: 20:37:15 | list-pods | Skipping deletion of user-supplied namespace: default
		=== CONT  kuttl
		    harness.go:376: run tests finished
		    harness.go:480: cleaning up
		    harness.go:532: removing temp folder: ""
		--- PASS: kuttl (0.87s)
		    --- PASS: kuttl/harness (0.00s)
		        --- PASS: kuttl/harness/list-pods (0.36s)
		PASS

thats an example of a 'passing' log, not much value perhaps, my vote would be to just show this log when there is a failure (e.g. no test cases).  If it turns out this log output is useful, we could add that on as a future feature later on.

File: images/scorecard-test-kuttl/cmd/test-kuttl/main.go (hunk 1 above)
Comment (jmccormick2001) [reply images/scorecard-test-kuttl/cmd/test-kuttl/main.go]: updated the code to use ioutil.ReadFile, good catch.

File: images/scorecard-test-kuttl/cmd/test-kuttl/main.go (hunk 2 above)
Comment (jmccormick2001) [reply images/scorecard-test-kuttl/cmd/test-kuttl/main.go]: I updated the code to only populate the Log when a kuttl error scenario is detected (e.g. len(tc) == 0).

**Pull request merged**